openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -59,6 +61,47 @@ const Carousel = {
             slidesToScroll: a,
             responsive: responsive_settings
         });
+
+        var addWork = function(work) {
+            return '<div class="book carousel__item slick-slide slick-active" aria-hidden="false" tabindex="-1" role="option" aria-describedby="slick-slide00" style="width: 128px;">' +
+                '<div class="book-cover">' +
+                  '<a href="' + work.key + '" tabindex="0">' +
+                    '<img class="bookcover" width="130" height="200" title="' + work.title + '" ' +
+                         'src="//covers.openlibrary.org/b/id/' + work.cover_id + '-M.jpg">' +
+                  '</a>' +
+                '</div>' +
+                '<div class="book-cta">' + // needs to be computed based on availability
+                  '<a class="btn primary " href="/books/OL1024614M/x/borrow" data-ol-link-track="subjects" ' +
+                  'title="Borrow eBook Comet" data-key="subjects" data-ocaid="comet00saga_1" tabindex="0">Borrow</a>' +
+                '</div>' +
+              '</div>';
+        }
+
+        $('.carousel-'+key).on('afterChange', function(e, slick, cur) {
+            var totalSlides = $('.carousel-' + key + '.slick-slider').slick("getSlick").$slides.length;
+            var numActiveSlides = $('.carousel-' + key + ' .slick-active').length;
+            var currentLastSlide = $('.carousel-' + key + '.slick-slider').slick('slickCurrentSlide') + numActiveSlides;
+            // this allows us to pre-load before hitting last page
+            var lastSlideOnSecondToLastPage = (totalSlides - numActiveSlides);
+
+            if (loadMoreUrl && (currentLastSlide >= lastSlideOnSecondToLastPage)) {
+                var limit = numActiveSlides * 3;
+                var url = loadMoreUrl + '?offset=' + totalSlides + '&limit=' + limit;
+                console.log(url);
```

Comment:
You can use this url to see what api we're hitting, though we should remove this console.log when we're doing :)